scripts/build.js

```diff
@@ -0,0 +1,172 @@
+  const args = process.argv.slice(2);
+
+  if (args.includes('--assets-only')) {
+    const assetDirs = ['chrome-enterprise', 'chrome-consumer', 'firefox-consumer', 'thunderbird-consumer'].map(dir => path.resolve(BUILD_DIRECTORY, dir));
+    await Promise.all(assetDirs.map(synchronizeFiles));
+    return;
+  }
+
+  if (args.includes('--incremental')) {
+    // Remove specific directories for incremental build
+    ['firefox-consumer', 'thunderbird-consumer', 'chrome-consumer', 'chrome-enterprise', 'generic-extension-wip/js/content_scripts'].forEach(dir =>
+      fs.removeSync(path.join(BUILD_DIRECTORY, dir))
+    );
+
+    fs.ensureDirSync(BUILD_DIRECTORY);
+    buildTS(false, './tsconfig.json', true, `${BUILD_DIRECTORY}/tsconfig.tsbuildinfo`);
+    buildTS(false, './conf/tsconfig.content_scripts.json', true, `${BUILD_DIRECTORY}/tsconfig.content_scripts.tsbuildinfo`);
+    buildTS(false, './conf/tsconfig.streams.json', true, `${BUILD_DIRECTORY}/tsconfig.streams.tsbuildinfo`);
+  } else {
+    fs.removeSync(BUILD_DIRECTORY);
+    runCmd('npx tsc --project ./conf/tsconfig.tooling.json');
+    fs.ensureDirSync(OUTPUT_DIRECTORY);
+    buildTS(true, './tsconfig.json');
+    buildTS(true, './conf/tsconfig.content_scripts.json');
+    buildTS(true, './conf/tsconfig.streams.json');
+  }
+
+  copyDependencies();
+
+  // patch imports with .js, e.g. replace './streams' with './streams.js'
+  // until https://github.com/openpgpjs/web-stream-tools/pull/20 is resolved
+  const streamDir = path.join(OUTPUT_DIRECTORY, 'lib/streams');
+  const streamFiles = fs.readdirSync(streamDir).map(file => path.join(streamDir, file));
+  applyRegexReplace(/'(.\/(streams|util|writer|reader|node-conversions))'/g, "'$1.js'", streamFiles);
+
+  // patch isUint8Array until https://github.com/openpgpjs/web-stream-tools/pull/23 is resolved
+  // First patch: replaces `return Uint8Array.prototype.isPrototypeOf(input);` with fallback to globalThis
+  applyRegexReplace(
+    /(\s*)return Uint8Array\.prototype\.isPrototypeOf\(input\);/g,
+    '$1return Uint8Array.prototype.isPrototypeOf(input) || globalThis.Uint8Array.prototype.isPrototypeOf(input);',
+    [...streamFiles, path.join(OUTPUT_DIRECTORY, 'lib/openpgp.js')]
+  );
+
+  // the following patches are until https://github.com/openpgpjs/openpgpjs/issues/1648 is fixed
+
+  // handles patterns like (n instanceof Uint8Array) or (arguments[i] instanceof Uint8Array)
+  // to replace them with (n instanceof Uint8Array || n instanceof globalThis.Uint8Array)
+  applyRegexReplace(/\(([^()\s]+) instanceof Uint8Array\)/g, '($1 instanceof Uint8Array || $1 instanceof globalThis.Uint8Array)', [
+    path.join(OUTPUT_DIRECTORY, 'lib/openpgp.js'),
+  ]);
+
+  // handles direct `return something instanceof Uint8Array;` expressions
+  applyRegexReplace(/return ([^()\s]+) instanceof Uint8Array;/g, 'return ($1 instanceof Uint8Array || $1 instanceof globalThis.Uint8Array);', [
+    path.join(OUTPUT_DIRECTORY, 'lib/openpgp.js'),
+  ]);
```

Comment:
looks like these patches can be removed, as original issue in `openpgpjs` marked as fixed

Reply:
Sounds good!